core/src/test/java/ai/timefold/solver/core/impl/domain/variable/listener/support/VariableListenerSupportTest.java

```diff
@@ -292,20 +292,18 @@ void shadowVariableListGraphEvents() {
         for (var visit : solution.getValues()) {
 
             // If a visit does not have a concurrent group, all variables of that visit share the same node.
-            if (visit.getConcurrentValueGroup() != null) {
-                // serviceStartTime and serviceFinishTime are in the same group, so no edge between them!
-                verifyAddEdge.accept(serviceReadyTime, visit, serviceStartTime, visit);
+            if (visit.getConcurrentValueGroup() != null && visit.equals(visit.getConcurrentValueGroup().get(0))) {
```

Comment:
I see a variant of this code already existed before. How did the groups work before this PR?

Also, arguably this now deserves a better name. (But the `groupKey` naming is undecided yet.)

Reply:
So, naming... "Group Key" arguably misses some context (group for what, what does the text mean), while including information that's obvious (it's a key). I'm currently thinking 

    "updateGroupSupplierName" (to mimic "supplierName" on @ShadowVariable)

The group is for updates, it references a supplier, by name.
I'm not happy with this name though - "updateGroup" doesn't say anything about the expected behavior either. Things other than "update" on my mind: deduplication, merging.

Any other suggestions?